.github/doctum/apiclient-services/doctum.js.twig

```diff
@@ -0,0 +1,353 @@
+        var versionSwitcher = document.getElementById('version-switcher');
+        if (versionSwitcher !== null) {
+            var currentVersion = versionSwitcher.options[versionSwitcher.selectedIndex].dataset.version;
+            versionSwitcher.addEventListener('change', function (event) {
+                var targetVersion = event.target.options[event.target.selectedIndex].dataset.version;
+                var candidateUrl = window.location.pathname.replace(currentVersion, targetVersion);
+                // Check if the page exists before redirecting to it
+                var testRequest = new XMLHttpRequest();
+                testRequest.open('HEAD', candidateUrl, false);
+                testRequest.send();
+                if (testRequest.status < 200 || testRequest.status > 399) {
+                    window.location = candidateUrl;
+                } else {
+                    // otherwise reroute to the home page of the new version
+                    window.location = this.value;
+                }
+            });
+        }
+        {% endif %}
+        {% verbatim -%}
+        Doctum.listenersRegistered = true;
+    },
+    loadTree: function () {
+        if (Doctum.treeLoaded) {
+            // Quick exit, already registered
+            return;
+        }
+        Doctum.rootPath = document.body.getAttribute('data-root-path');
+        Doctum.buildTreeNode(Doctum.treeJson.tree, document.getElementById('api-tree'), Doctum.treeJson.treeOpenLevel);
+
+        // Toggle left-nav divs on click
+        $('#api-tree .hd span').on('click', function () {
+            $(this).parent().parent().toggleClass('opened');
+        });
+
+        // Expand the parent namespaces of the current page.
+        var expected = $('body').attr('data-name');
+
+        if (expected) {
+            // Open the currently selected node and its parents.
+            var container = $('#api-tree');
+            var node = $('#api-tree li[data-name="' + expected + '"]');
+            // Node might not be found when simulating namespaces
+            if (node.length > 0) {
+                node.addClass('active').addClass('opened');
+                node.parents('li').addClass('opened');
+                var scrollPos = node.offset().top - container.offset().top + container.scrollTop();
+                // Position the item nearer to the top of the screen.
+                scrollPos -= 200;
+                container.scrollTop(scrollPos);
+            }
+        }
+        Doctum.treeLoaded = true;
+    },
+    pagePartiallyLoaded: function (event) {
+        Doctum.initListeners();
+        Doctum.loadTree();
+        Doctum.loadAutoComplete();
+    },
+    pageFullyLoaded: function (event) {
+        // it may not have received DOMContentLoaded event
+        Doctum.initListeners();
+        Doctum.loadTree();
+        Doctum.loadAutoComplete();
+        // Fire the event in the search page too
+        if (typeof DoctumSearch === 'object') {
```

Comment:
This sounds off to me. Almost everything is type `object` in JS. If you type on the console the next two examples, both of them are true:
```javascript
let a = [];
typeof a === 'object' // true
```
```javascript
let a = {};
typeof a === 'object' // true
```

Reply:
The first comment explains this